From: Buildroot compatibility
Subject: [PATCH] dtc-lexer: declare yylloc as extern

GCC 10+ defaults to -fno-common, so defining YYLTYPE yylloc in both
dtc-lexer.l and the bison-generated parser causes a link error:

  multiple definition of `yylloc'

Declare it as extern in the lexer; the parser provides the definition.

Signed-off-by: local <local@local>
---
 dtc-lexer.l | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dtc-lexer.l b/dtc-lexer.l
index 1111111..2222222 100644
--- a/dtc-lexer.l
+++ b/dtc-lexer.l
@@ -38,7 +38,7 @@ LINECOMMENT	"//".*\n
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
+extern YYLTYPE yylloc;
 extern bool treesource_error;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
-- 
2.25.1
